### ISSUE TITLE
GurobiError(10018,"Problem adding constraints") + CPLEX segfault

### ISSUE THREAD

**Issue by hunsa** [JuliaOpt/JuMP.jl#364]
I have been playing with Gurobi, CPLEX and JuMP. At least CPLEX, had worked before (before upgrading the OS, julia, JuMP, CPLEX).

My config looks like this:
Julia Version 0.3.5
Commit a05f87b (2015-01-08 22:33 UTC)
"JuMP"                  => v"0.7.3"
"CPLEX"                 => v"0.0.11"
"Gurobi"                => v"0.1.23"

Gurobi
---------
I have Gurobi 6.0.0 build v6.0.0rc2 (mac64) 
If I change the solver in bnatt350.jl to 

    using Gurobi
    mod = Model(solver=GurobiSolver(PreCrush=1, Cuts=0, Presolve=0, DisplayInterval=1))
    m_internal = MathProgBase.model(GurobiSolver())

I get
```
Read MPS format model from file /Users/sascha/Downloads/JuMP.jl-master/examples/data/bnatt350.mps
Reading time = 0.01 seconds
bnatt350: 4923 rows, 3150 columns, 19061 nonzeros
ERROR: GurobiError(10018,"Problem adding constraints")
 in include at /opt/local/lib/julia/sys.dylib
 in include_from_node1 at /opt/local/lib/julia/sys.dylib
while loading /Users/sascha/Downloads/JuMP.jl-master/examples/bnatt350.jl, in expression starting on line 70
```

CPLEX
----------
CPLEX_Studio1261 (also Mac x86_64)

```
Selected objective sense:  MINIMIZE
Selected objective  name:  obj
Selected RHS        name:  rhs
Selected bound      name:  bnd
Parallel mode: deterministic, using up to 8 threads for concurrent optimization.
Initializing dual steep norms . . .

Iteration log . . .
Iteration:     1   Dual objective     =             0.000000
Perturbation started.
Iteration:   101   Dual objective     =             0.000000
Iteration:   253   Dual objective     =             0.000020

signal (11): Segmentation fault: 11
_84c3c4d55d10ed85d167906892b45d7d at /Users/sascha/local/CPLEX_Studio1261/cplex/bin/x86-64_osx/libcplex1261.dylib (unknown line)
_86ca56ee2db9b66412fe44a38f0981bc at /Users/sascha/local/CPLEX_Studio1261/cplex/bin/x86-64_osx/libcplex1261.dylib (unknown line)
```
..... (I cut here)

Interestingly, both solvers seem to work fine with the other examples.
It would be great if someone could look into this.

-Sascha

**Comment by mlubin** [comment]
I can't reproduce this on Linux with Gurobi 6 and CPLEX 12.6.1. Looks like an OS X specific issue, @joehuchette?

**Comment by joehuchette** [comment]
I can't reproduce the behavior with CPLEX 12.6 and CPLEX.jl v0.0.11, nor with Gurobi 6.0.0 and Gurobi.jl v0.1.23 (JuMP master, but I doubt that is the difference). Ran it with both julia 0.3.5 and 0.4, didn't make a difference. This is bizarre.

**Comment by joehuchette** [comment]
(I did notice that Gurobi 6 now accepts the heuristic solution, which 5.6 didn't for some reason. I complained on the mailing list and maybe they listened?)

**Comment by joehuchette** [comment]
@hunsa Do you have an older version of CPLEX by chance? Maybe the segfault is caused by some incompatibility introduced between 12.6 and 12.6.1?

**Comment by hunsa** [comment]
I had noticed the error with CPLEX 12.5 and JuMP first.
Then, I thought I should update CPLEX and since it wasn't working, I tried Gurobi.

**Comment by mlubin** [comment]
@hunsa, if you have access to a different computer with similar OS, could you see if the issue is reproducible?

**Comment by hunsa** [comment]
Here is one example that also triggers the problem.

```
using JuMP
using MathProgBase
#using GLPKMathProgInterface
using CPLEX
#using Gurobi
using Debug


@debug function solve_mip()

    srand(1)

#    m = Model()
    m = Model(solver=CplexSolver(CPX_PARAM_SCRIND=0))
#    m = Model(solver=GurobiSolver())

    n = 25
    a = zeros(n)

    min_tasks = 10

    for i=1:n
        a[i] = rand(1:10)
    end

    @defVar(m, x[1:n], Bin )

    @defVar(m, t1 >= 0 )
    @defVar(m, dummy >= 0 )
    @setObjective(m, Min, dummy)

    @addConstraint(m, sum{ a[j] * x[j], j=1:n } <= dummy )
    @addConstraint(m, sum{ x[j], j=1:n } >= min_tasks )

    status = solve(m)

    println("Objective value: ", getObjectiveValue(m))

    x = getValue(x)
    print(x)

end

solve_mip()
```

Gurobi and CPLEX work if I have either 

    @addConstraint(m, sum{ a[j] * x[j], j=1:n } <= dummy )

or

    @addConstraint(m, sum{ x[j], j=1:n } >= min_tasks )

defined.
With both constraints, I get the errors shown above.

**Comment by hunsa** [comment]
I will try on my Linux box tomorrow to verify.

**Comment by mlubin** [comment]
Probably a silly question, but does the issue also occur when ``@debug`` is removed?

**Comment by hunsa** [comment]
Sure, I just thought I should debug before I ask questions here. Then, I was "happy" that even your examples triggered that problem. So, I could be kind of sure that it wasn't my code. Anyway, @debug makes no difference.

**Comment by joehuchette** [comment]
FWIW I can't replicate the error with julia 0.3.5, either with CPLEX or Gurobi. There's [this thread](https://groups.google.com/forum/#!topic/gurobi/HcqOabVFq8o) that talks about the particular Gurobi error code; maybe there's a bug in the code that constructs the sparse representations of the rows?

**Comment by mlubin** [comment]
Another thing to try is removing or renaming the ``~/.julia`` directory and reinstalling the packages from scratch, in case there are any local modifications or strange version mismatches going on.

(By the way, be sure to escape `@debug` inside ticks so we don't bother @debug any more.)

**Comment by hunsa** [comment]
I ran it on Linux and the latest versions of Gurobi and CPLEX, and it worked without problems.
Then I uninstalled julia (which was provided by macports) and recompiled julia from the latest tarball (same version though). Now, the solvers (CPLEX and Gurobi) also work nicely on OS X. 
Thanks for the suggestions. It is still weird that this problem (with the macports version) only occurs with some type of constraints in these particular cases.
